azurerm/internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -171,6 +172,37 @@ func resourceArmMsSqlVirtualMachine() *schema.Resource {
 				ValidateFunc: validate.MsSqlVMLoginUserName,
 			},
 
+			"storage_configuration_settings": {
```

Comment:
I think settings is redundant here
```suggestion
			"storage_configuration": {
```